### ISSUE TITLE
Request: Filter lists for MyReadingManga

### ISSUE THREAD

**Issue by happywillow0** [inorichi/tachiyomi-extensions#1354]
Is it possible to add a genre selector to the MyReadingMannga extension? 

While it is possible to use the search function to browse a specific tag or genre, the page feature doesn't currently work (I'm thinking of the infinite load search feature rather than pages)

On the other hand, the website uses the pretty simple url scheme for the options with

- base/lang/english/
- base/genre/
- base/tag/

And these page loads the same layout as the home page which should use the same scraping as the popularmanaga

However, while trying to try it myself, I realized that the filter lists run though the search function and the search layout is different than the other pages. Thus, I'm not sure if the lists will be really possible to implement easily. 

Any additional thoughts are appreciated and I understand if it is not possible with how the app is set up.

**Comment by happywillow0** [comment]
I got an updated search function working. However, when I try to replace /search/ with something else to test such as /genre/, I get a http 503 (or was it a 506) error. I'm not sure if I'm just doing something wrong or maybe it is a cloudflare error.

**Comment by waptik** [comment]
code snippets???

**Comment by happywillow0** [comment]
It's been a while since I looked at this. I used what I learned from this so far to "re-work" the search code in general. Although filter lists are not in yet, at least you can search by tag / genre now. 

The part I was editing and testing with is the following search block. Here is the current code:
```
 override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
        val query2 = URLEncoder.encode(query, "UTF-8")
        val uri = Uri.parse("$baseUrl/search/").buildUpon()
                .appendEncodedPath(query2)
                .appendPath("page")
                .appendPath("$page")
        return GET(uri.toString())
    }
```
My previous tests, this works:
```
override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
        val query2 = URLEncoder.encode(query, "UTF-8")
        val uri = Uri.parse("$baseUrl/genre/yaoi/").buildUpon()
        return GET(uri.toString())
    }
```
but this doesn't and errors 503 if I remember correctly :
```
override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
        val query2 = URLEncoder.encode(query, "UTF-8")
        val uri = Uri.parse("$baseUrl/genre/").buildUpon()
                .appendPath("yaoi")
        return GET(uri.toString())
    }
```

What I figure, once I can get the `appendpath `working correctly, I can start trying to build the actual list code out.

**Comment by happywillow0** [comment]
#1428 submitted. This puts in my initial request for basic genre lists. 

This can be further enhanced by:

- Adding a filter for Type (eg Oneshot, Manga,  Manhua, Manhwa, etc)
- Parsing a live genre list from the website (not that I'm expecting it to actually change)
- Adding in and possibly parsing the "Popular Tags"
- Parsing and adding the Categories list

**Comment by waptik** [comment]
Sorry for my late reply. I've seen your pull and went through the codes.
Before sending a PR, i take it that it worked for you after testing it locally, right?
What does buildupon do? How do a search url query and a filter look like?

**Comment by SnakeDoc83** [comment]
You're right, I forgot companion objects don't play well with variables from a class.

**Comment by happywillow0** [comment]
I know people are busy with real lives (including me) and the website updates as well. I managed to pull out the block of code outside the companion object and I have updated the PR request. 

Filters are now grabbed by a OkHttpClient call and parsing the home page for the genres, popular tags, and category list. 

I've been running the built extension for a while now and everything is working fine for me. 

Thank you guys for the help.
